plugins/apitracing/src/lib/os/Extractor.cpp

```diff
@@ -0,0 +1,316 @@
+                // to the nexţ iteration
+                VmiCore::addr_t structPointer = dereferencePointer(address, cr3);
+                ExtractedParameterInformation extraction{.name = parameter.name, .data = {}, .backingParameters = {}};
+                extraction.backingParameters =
+                    extractBackingParameters(parameter.backingParameters, structPointer, cr3);
+                extractedBackingParameters.emplace_back(extraction);
+            }
+        }
+        return extractedBackingParameters;
+    }
+
+    BasicTypes Extractor::resolveBasicType(std::string_view basicType)
+    {
+        if (basicType == "LPSTR_32")
+        {
+            return BasicTypes::LPSTR_32;
+        }
+        if (basicType == "LPSTR_64")
+        {
+            return BasicTypes::LPSTR_64;
+        }
+        if (basicType == "LPWSTR_32")
+        {
+            return BasicTypes::LPWSTR_32;
+        }
+        if (basicType == "LPWSTR_64")
+        {
+            return BasicTypes::LPWSTR_64;
+        }
+        if (basicType == "UNICODE_WSTR_32")
+        {
+            return BasicTypes::UNICODE_WSTR_32;
+        }
+        if (basicType == "UNICODE_WSTR_64")
+        {
+            return BasicTypes::UNICODE_WSTR_64;
+        }
+        if (basicType == "__ptr32")
+        {
+            return BasicTypes::__PTR32;
+        }
+        if (basicType == "__ptr64")
+        {
+            return BasicTypes::__PTR64;
+        }
+        if (basicType == "int")
+        {
+            return BasicTypes::INT;
+        }
+        if (basicType == "long")
+        {
+            return BasicTypes::LONG;
+        }
+        if (basicType == "__int64")
+        {
+            return BasicTypes::__INT64;
+        }
+        if (basicType == "unsigned __int64")
+        {
+            return BasicTypes::UNSIGNED___INT64;
+        }
+        if (basicType == "unsigned long")
+        {
+            return BasicTypes::UNSIGNED_LONG;
+        }
+        if (basicType == "unsigned int")
+        {
+            return BasicTypes::UNSIGNED_INT;
+        }
+        if (basicType == "unsigned short")
+        {
+            return BasicTypes::UNSIGNED_SHORT;
+        }
+        return BasicTypes::INVALIC_BASIC_TYPE;
+    }
+
+    uint64_t Extractor::readParameter(uint64_t stackPointerVA, uint8_t parameterSize, uint64_t cr3) const
+    {
+        auto output = introspectionAPI->read64VA(stackPointerVA, cr3);
+        return zeroGarbageBytes(output, parameterSize);
+    }
+
+    uint64_t Extractor::zeroGarbageBytes(uint64_t parameter, uint8_t parameterSize) const
+    {
+        parameter = parameter << (addressWidth - parameterSize * ConstantDefinitions::byteSize);
+        parameter = parameter >> (addressWidth - parameterSize * ConstantDefinitions::byteSize);
+        return parameter;
+    }
+
+    std::string Extractor::extractString(VmiCore::addr_t stringPointer, uint64_t cr3) const
```

Comment:
nit: This function is just boiler plate and calls to it can be replaced by a simple `*`

Reply:
yeah, but with libvmi not accepting our wstring extraction there will a more complex extract function. So not to mix abstractions layers I would keep it as is.